pkg/ebpf/cached_conntrack.go

```diff
@@ -0,0 +1,164 @@
+// +build linux_bpf
+
+package ebpf
+
+import (
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	"github.com/DataDog/datadog-agent/pkg/network/netlink"
+	"github.com/DataDog/datadog-agent/pkg/process/util"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	ct "github.com/florianl/go-conntrack"
+	"github.com/golang/groupcache/lru"
+	"golang.org/x/sys/unix"
+)
+
+type refCountedConntrack struct {
+	ctrk  netlink.Conntrack
+	count *uint32
+}
+
+type cachedConntrack struct {
+	sync.Mutex
+	procRoot         string
+	cache            *lru.Cache
+	conntrackCreator func(int) (netlink.Conntrack, error)
+	closed           bool
+}
+
+func newCachedConntrack(procRoot string, conntrackCreator func(int) (netlink.Conntrack, error), size int) *cachedConntrack {
+	cache := &cachedConntrack{
+		procRoot:         procRoot,
+		conntrackCreator: conntrackCreator,
+		cache:            lru.New(size),
+	}
+
+	cache.cache.OnEvicted = func(_ lru.Key, v interface{}) {
+		v.(refCountedConntrack).Decr()
```

Comment:
sounds look like you're concerned about calling `netlink.Conntrack.Close()`  when you have a request in flight. IMO `netlink.Conntrack` should abstract that

Reply:
Abstracting it in `netlink` would mean I would have to move the cache also to completely hide that reference counting is going on. I opted to have the wrapper implement `netlink.Conntrack` here, whose `Close` does decrement the ref count. Also cleans up some of the references to ref counting in `Exists`.